UefiTestingPkg/FunctionalSystemTests/MemoryProtectionTest/App/MemoryProtectionTestApp.c

```diff
@@ -964,17 +1008,15 @@ UefiPoolGuard (
   EFI_STATUS                      Status;
   UINTN                           AllocationSize;
   UINT8                           Index = 0;
+  UINT64                          Attributes;
+  EFI_PHYSICAL_ADDRESS            PoolGuard;
 
   DEBUG ((DEBUG_INFO, "%a - Testing Type: %a\n", __FUNCTION__, MEMORY_TYPES[MemoryProtectionContext.TargetMemoryType]));
 
-  if (MemoryProtectionContext.DynamicActive) {
-    if (mNonstopModeProtocol == NULL) {
-      UT_ASSERT_NOT_EFI_ERROR (GetNonstopProtocol ());
-    }
+  if (MemoryProtectionContext.TestingMethod == MemoryProtectionTestMemoryAttributeProtocol) {
+    UT_ASSERT_NOT_NULL (mMemoryAttributeProtocol);
 
     for (Index = 0; Index < NUM_POOL_SIZES; Index++) {

```

Comment:
Was there a reason to define `NUM_POOL_SIZES` instead of using `ARRAY_SIZE()`? Array size, would help keep array length and the max index consistent.

Reply:
NUM_POOL_SIZES has been used in the test since 2018 when Bret originally wrote it but I agree using array size also works

Reply:
I saw that too; I'm not requesting you to change it. For some of these comments, your feedback based on your experience in the code is helpful to determine if we should change it at some point.

Reply:
Removed NUM_POOL_SIZES